lib/CGI.pm

```diff
@@ -649,7 +641,15 @@ sub init {
 	  # Uncomment this line to have the contents of the query string
 	  # APPENDED to the POST data.
 	  # $query_string .= (length($query_string) ? '&' : '') . $ENV{'QUERY_STRING'} if defined $ENV{'QUERY_STRING'};
-	  last METHOD;
+	  last METHOD if (defined $query_string && length $query_string);
+      }
+
+      # If nothing found, let's try to get data from request parameters
+      if ($is_xforms || $meth=~/^(GET|HEAD|DELETE|POST|PUT)$/) {
```

Comment:
Again as above - this is almost certainly going to break backwards compatibility for some users.